libraries/core/src/main/java/com/paritytrading/philadelphia/FIXConnection.java

```diff
@@ -387,7 +395,8 @@ public void keepAlive() throws IOException {
      */
     @Override
     public void close() throws IOException {
-        channel.close();
+        readableChannel.close();
+        writableChannel.close();
     }
```

Comment:
These are iffy - ideally the second close should have been in finally, or should have caught the 2 exceptions and thrown another one with those added as suppressed.

Reply:
Yes, let's wrap this so that both methods get called even if the first throws an exception. Maybe something like this could work? 🤔 
```java
IOException readableException = null;
IOException writableException = null;

try {
    readableChannel.close();
} catch (IOException e) {
    readableException = e;
}

try {
    writableChannel.close();
} catch (IOException e) {
    writableException = e;
}

if (writableException != null) {
    if (readableException != null)
        writableException.addSuppressed(readableException);

    throw writableException;
}
else if (readableException != null) {
    throw readableException;
}
```

Reply:
How about

```
        try (Closeable ignored1 = rxChannel; Closeable ignored2 = txChannel) {
            // we just want to close both channels reliably  
        }
```